server/tabby_server/services/library.py

```diff
@@ -15,151 +16,144 @@
 # Load the .env variables.
 load_dotenv()
 
+# Global Variables
+MAX_RESULTS = 40  # Determines number of reults Google Books Returns (max 40)
+API_KEY = str(os.getenv("API_KEY"))  # API Key to access Google Books
+
 # Blueprint to be accessible from __main__.py.
 # To call, use: "/library/<function_route>"
 books_api = Blueprint("library", __name__)
 
 
 # Dataclass to hold book attributes from the google books api call.
 @dataclass(frozen=True, kw_only=True)
-class BookAttr:
-    isbn: int
+class Book:
+    isbn: str
     title: str
     authors: list
     rating: str
-    summary: str
+    excerpt: str = field(init=False)  # Cannot be initialized
+    summary: str = ""
     thumbnail: str
     page_count: str
     categories: list
     publisher: str
     published_date: str
 
-
-def parameter_handler(param="", index=-1):
+    # After the dataclass is made, run this.
+    def __post_init__(self) -> None:
+        # Checks if summary is longer than 50 characters.
+        # If it is, save the first 47 and add "..." to excerpt.
+        # Otherwise, save the summary.
+        if len(self.summary) > 50:
+            object.__setattr__(self, "excerpt", f"{self.summary[:46]}...")
+        else:
+            object.__setattr__(self, "excerpt", self.summary)
```

Comment:
I think it's better to make this a property instead of an actual attribute:
```python
@property
def excerpt(self) -> str:
    return self.summary[:46]
```
That way, it's not modifiable; we know it's a calculated property.

Reply:
Also, indices don't need to be in bounds when taking a slice.